pyspark_datasources/robinhood.py

```
@@ -0,0 +1,284 @@
+
+
+@dataclass
+class CryptoPair(InputPartition):
+    """Represents a single crypto trading pair partition for parallel processing."""
+    symbol: str
+
+
+class RobinhoodDataReader(DataSourceReader):
+    """Reader implementation for Robinhood Crypto API data source."""
+
+    def __init__(self, schema: StructType, options: Dict):
```

Ruff (0.12.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)